inc/knowbaseitem.class.php

```diff
@@ -1208,50 +1232,73 @@ static function getListRequest(array $params, $type = 'search') {
                $addscore = [];
                if (KnowbaseItemTranslation::isKbTranslationActive()
                    && (countElementsInTable('glpi_knowbaseitemtranslations') > 0)) {
-                  $addscore = ['`glpi_knowbaseitemtranslations`.`name`',
-                                    '`glpi_knowbaseitemtranslations`.`answer`'];
+                  $addscore = [
+                     'glpi_knowbaseitemtranslations.name',
+                     'glpi_knowbaseitemtranslations.answer'
+                  ];
                }
-               $score = " ,(MATCH(`glpi_knowbaseitems`.`name`, `glpi_knowbaseitems`.`answer`)
-                           AGAINST('$search_wilcard' IN BOOLEAN MODE)";
+
+               $expr = "(MATCH(" . $DB->quoteName('glpi_knowbaseitems.name') . ", " . $DB->quoteName('glpi_knowbaseitems.answer') . ")
+                           AGAINST(" . $DB->quote($search_wilcard) . " IN BOOLEAN MODE)";
 
                if (!empty($addscore)) {
                   foreach ($addscore as $addscore_field) {
-                     $score.= " + MATCH($addscore_field)
-                                        AGAINST('$search_wilcard' IN BOOLEAN MODE)";
+                     $expr .= " + MATCH(" . $DB->quoteName($addscore_field) . ")
+                                        AGAINST('" . $DB->quoteName($search_wilcard) . " IN BOOLEAN MODE)";
                   }
                }
-               $score .=" ) AS SCORE ";
+               $expr .=" ) AS SCORE ";
+               $criteria['SELECT'][] = new QueryExpression($expr);
 
-               $where_1 = $where." AND (MATCH(`glpi_knowbaseitems`.`name`,
-                                             `glpi_knowbaseitems`.`answer`)
-                          AGAINST('$search_wilcard' IN BOOLEAN MODE) ";
+               $search_where = [
+                  new QueryExpression(
+                     "MATCH(" . $DB->quoteName('glpi_knowbaseitems.name') .",
+                        " . $DB->quoteName('glpi_knowbaseitems.answer') .")
+                        AGAINST(" . $DB->quote($search_wilcard) . " IN BOOLEAN MODE)"
+                  )
+               ];
 
                if (!empty($addscore)) {
                   foreach ($addscore as $addscore_field) {
-                     $where_1.= "OR $addscore_field IS NOT NULL
-                                    AND MATCH($addscore_field)
-                                        AGAINST('$search_wilcard' IN BOOLEAN MODE)";
+                     $search_where[] = [
+                        'OR'  => [
```

Comment:
This `OR` is at wrong place. Opeartor here should be `AND` (or should be removed).
What was done before was a `OR` between condition created on line 1253 and each condition created inside the foreach loop.
All of this was surrounded by the `(` present before the `MATCH` function and the `)` present after the foreach loop.